### PR TITLE
[ST-Engine] Use SeatunnelChildFirstClassLoader to loader connector plugin

### PR DESCRIPTION
**Pull request by EricJoy2048**
1、Finish a main process, submit-scheduler-finish.
2、Add engine e2e.

### CODE REVIEW

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      parallelism = 3
+    }
+
+  # If you would like to get more information about how to configure seatunnel and see full list of source plugins,
+  # please go to https://seatunnel.apache.org/docs/flink/configuration/source-plugins/Fake
+}
+
+transform {
+  # If you would like to get more information about how to configure seatunnel and see full list of transform plugins,
+  # please go to https://seatunnel.apache.org/docs/flink/configuration/transform-plugins/Sql
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf:43]: Use `https://seatunnel.apache.org/docs/transform/sql` ?

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    file_format="text"
+    sink_columns=["name","age"]
+    filename_time_format="yyyy.MM.dd"
+    is_enable_transaction=true
+    save_mode="error"
+
+  }
+
+  # If you would like to get more information about how to configure seatunnel and see full list of sink plugins,
+  # please go to https://seatunnel.apache.org/docs/flink/configuration/sink-plugins/Console
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf:64]: Use `https://seatunnel.apache.org/docs/connector-v2/sink/LocalFile` ?

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file_complex.conf [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    file_format="text"
+    sink_columns=["name","age"]
+    filename_time_format="yyyy.MM.dd"
+    is_enable_transaction=true
+    save_mode="error",
+    source_table_name="fake"
+  }
+
+  # If you would like to get more information about how to configure seatunnel and see full list of sink plugins,
+  # please go to https://seatunnel.apache.org/docs/flink/configuration/sink-plugins/Console
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file_complex.conf:70]: as above

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServer.java [hunk 4]
```diff
@@ -65,7 +65,7 @@ public SeaTunnelServer(@NonNull Node node, @NonNull SeaTunnelConfig seaTunnelCon
         this.liveOperationRegistry = new LiveOperationRegistry();
         this.seaTunnelConfig = seaTunnelConfig;
         this.executorService =
-            Executors.newFixedThreadPool(seaTunnelConfig.getEngineConfig().getServerExecutorPoolSize());
+            Executors.newCachedThreadPool();
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServer.java]: Define thread name?

explame:
```java
Executors.newCachedThreadPool(new ThreadFactoryBuilder()
                .setNameFormat("seatunnel-server-executor-%d").build());
```

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/SimpleResourceManager.java [hunk 5]
```diff
@@ -40,7 +40,7 @@ public Address applyForResource(long jobId, long taskId) {
             Map<Long, Address> jobAddressMap = physicalVertexIdAndResourceMap.computeIfAbsent(jobId, k -> new HashMap<>());
 
             Address localhost =
-                    jobAddressMap.putIfAbsent(taskId, new Address("localhost", 5701));
+                    jobAddressMap.putIfAbsent(taskId, new Address("192.168.1.10", 5801));
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/SimpleResourceManager.java]: Should not commit local ip?

File: seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/TaskExecutionServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+//import java.util.Collections;
+//import java.util.List;
+//import java.util.concurrent.CompletableFuture;
+//import java.util.concurrent.CopyOnWriteArrayList;
+//import java.util.concurrent.ExecutionException;
+//import java.util.concurrent.TimeUnit;
+//import java.util.concurrent.atomic.AtomicBoolean;
+//import java.util.function.BiConsumer;
+//
+//public class TaskExecutionServiceTest {
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/TaskExecutionServiceTest.java]: Don't commit this file?

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf (hunk 1 above)
Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf:43]: done

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf (hunk 2 above)
Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf:64]: done

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file_complex.conf (hunk 3 above)
Comment (EricJoy2048) [reply seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file_complex.conf:70]: done

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServer.java (hunk 4 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/SeaTunnelServer.java]: done

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/SimpleResourceManager.java (hunk 5 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/resourcemanager/SimpleResourceManager.java]: done

File: seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/TaskExecutionServiceTest.java (hunk 6 above)
Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/TaskExecutionServiceTest.java]: done

**Review by Hisoka-X — APPROVED** [review]
LGTM

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  job.mode = "STREAMING"
+  execution.checkpoint.interval = 5000
+  #execution.checkpoint.data-uri = "hdfs://localhost:9000/checkpoint"
+}
+
+source {
+  # This is a example source plugin **only for test and demonstrate the feature source plugin**
+    FakeSource {
+      result_table_name = "fake"
+      field_name = "name,age",
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf:33]: Use new config style?   #2406

example:
```xml
FakeSource {
    result_table_name = "fake"
    fields {
        name = string
        age = int
    }
}
```

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+# limitations under the License.
+#
+######
+###### This config file is a demonstration of streaming processing in seatunnel config
+######
+
+env {
+  # You can set flink configuration here
+  execution.parallelism = 1
+  job.mode = "STREAMING"
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file.conf:24]: will it stop automatically？

File: seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file_complex.conf [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+      result_table_name = "fake"
+      field_name = "name,age",
+      parallelism = 3
+    }
+
+    FakeSource {
+      result_table_name = "fake"
+      field_name = "name,age",
+      parallelism = 3
+    }
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-engine-e2e/src/test/resources/fakesource_to_file_complex.conf:41]: duplicate define？

**Pull request merged**